spec/struct.dd

```diff
@@ -1363,6 +1363,156 @@ $(H3 $(LNAME2 implicit-copy-constructors, Implicit Copy Constructors))
     $(P If the generated copy constructor fails to type check, it will receive the `@disable` attribute.)
 
 
+$(H2 $(LEGACY_LNAME2 StructMoveConstructor, struct-move-constructor, Struct Move Constructors))
+
+    $(P Move constructors are used to initialize a `struct` instance from
+    another instance of the same type, and then the other instance is reset to its initial state.
+    This effects a move rather than a copy. A `struct` that defines a move constructor
+    is not $(RELATIVE_LINK2 POD, POD).)
+
+    $(P A constructor declaration is a move constructor declaration if it meets
+    the following requirements:)
+
+    $(UL
+    $(LI The `this` part of the declaration is preceded by an `=`, i.e. `=this` to distinguish
+    the move constructor from other constructors.)
+
+    $(LI It takes exactly one parameter without a
+    $(DDSUBLINK spec/function, function-default-args, default argument),
+    followed by any number of parameters with default arguments.)
+
+    $(LI Its first parameter is a $(B not)
+    $(DDSUBLINK spec/function, ref-params, `ref` parameter).)
+
+    $(LI The type of its first parameter is the same type as
+    $(DDSUBLINK spec/type, typeof, `typeof(this)`), optionally with one or more
+    $(DDLINK spec/const3, Type Qualifiers, type qualifiers) applied to it.)
+
+    $(LI It is not a
+    $(DDSUBLINK spec/template, template_ctors, template constructor declaration).)
+    )
+
+    ---
+    struct A
+    {
+        =this(return scope A rhs) {}                        // move constructor
+        =this(return scope const A rhs, int b = 7) {}       // move constructor with default parameter
+    }
+    ---
+
+    $(P The move constructor is type checked as a normal constructor.)
+
+    $(P If a move constructor is defined, implicit calls to it will be inserted
+    in the following situations:)
+
+    $(OL
+    $(LI When an rvalue is used to initialize a variable:)
```

Comment:
What about `A a = A();` - the RHS is an rvalue, but presumably the move constructor is not called?